q2_demux/_summarize/_visualizer.py

```diff
@@ -103,7 +114,9 @@ def _compute_stats_of_df(df):
     return df_stats
 
 
-def summarize(output_dir: str, data: _PlotQualView, n: int=10000) -> None:
+def summarize(output_dir: str, data: _PlotQualView,
+              n: int=10000, delta=1) -> None:
+    delta = abs(int(round(delta)))
```

Comment:
Instead of doing all of this, you can rely on the qiime2 types to guard the input for you. And if you need a positive value change the primitive type to be: `Int % Range(0, None)`.